ddtrace/_trace/_span_pointer.py

```diff
@@ -64,7 +64,7 @@ def __post_init__(self):
     def __repr__(self):
         return (
             f"SpanPointer(trace_id={self.trace_id}, span_id={self.span_id}, kind={self.kind}, "
-            f"direction={self.direction}, hash={self.hash}, attributes={self.attributes})"
+            f"direction={self.attributes.get('ptr.dir')}, hash={self.attributes.get('ptr.hash')}, attributes={self.attributes})"
```

Comment:
Can we update this test case to also set a span pointer. This will act as a regression test: https://github.com/DataDog/dd-trace-py/blob/main/tests/tracer/test_span.py#L869.